src/vs/workbench/contrib/tasks/node/tasks.ts

```diff
@@ -5,21 +5,42 @@
 
 import * as nls from 'vs/nls';
 
-import * as crypto from 'crypto';
-
 import * as Objects from 'vs/base/common/objects';
 
 import { TaskIdentifier, KeyedTaskIdentifier, TaskDefinition } from 'vs/workbench/contrib/tasks/common/tasks';
 import { TaskDefinitionRegistry } from 'vs/workbench/contrib/tasks/common/taskDefinitionRegistry';
 
 namespace KeyedTaskIdentifier {
-	export function create(value: TaskIdentifier): KeyedTaskIdentifier {
-		const hash = crypto.createHash('md5');
-		hash.update(JSON.stringify(value));
-		let result = { _key: hash.digest('hex'), type: value.taskType };
-		Objects.assign(result, value);
+	function hexValue(input: string): string {
+		let result: number = 0;
+		for (let i = 0; i < input.length; i++) {
+			result += input.charCodeAt(i);
+		}
+		return '' + result;
+	}
+
+	function sortedStringify(literal: any): string {
+		const keys = Object.keys(literal).sort();
+		let result: string = '';
+		for (let position in keys) {
+			let test = literal[keys[position]];
+			if (test instanceof Object) {
+				test = sortedStringify(test);
+			}
+			let stringified: string = '' + test;
+			if (stringified.length > 50) { // 50 is an arbitrary number
+				// This can result in collisions, but it's unlikely
+				stringified = hexValue(stringified);
```

Comment:
Getting the hexValue has too high a risk of collisions. I need to find something else.